### PR TITLE
Remove mark_cache_size default value of tiflash.toml after v5.4.0

### PR DESCRIPTION
**Pull request by hongyunyan**
### What problem does this PR solve? 
ref https://github.com/pingcap/tiflash/pull/6861

### What is changed and how it works?
Remove mark_cache_size default value of tiflash.toml after v5.4.0
Because tiflash add default_mark_cache_size = 5G in code from v5.4.0, so for the version after v5.4.0, we can directly remove this item in config.

### Check List 

Tests 

 - Unit test
 - Integration test
 - [x] Manual test (add detailed scripts or steps below)
   use tiup-playground and tiup-cluster to start cluster in nightly and v5.3.0 to check the tiflash.toml
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

Release notes:

```release-note
NONE
```

### CODE REVIEW

**Review by kaaaaaaang — APPROVED** [review]

File: pkg/cluster/spec/tiflash.go [hunk 1]
```diff
@@ -438,16 +439,17 @@ func (i *TiFlashInstance) initTiFlashConfig(ctx context.Context, clusterVersion
 	if tidbver.TiFlashNotNeedSomeConfig(clusterVersion) {
 		// For 5.4.0 or later, TiFlash can ignore application.runAsDaemon setting
```
Comment (breezewish) [inline pkg/cluster/spec/tiflash.go:440]: Please update the comment as well.

**Review by breezewish — APPROVED** [review]

File: pkg/cluster/spec/tiflash.go (hunk 1 above)
Comment (hongyunyan) [reply pkg/cluster/spec/tiflash.go]: done

**Pull request merged**